server/python_backend/main.py

```
@@ -120,6 +120,7 @@ class FilterRequest(BaseModel):
 
 # Dashboard Endpoints
 @app.get("/api/dashboard/stats", response_model=DashboardStatsResponse)
+@performance_monitor.track_function_performance
 async def get_dashboard_stats(request: Request, db: DatabaseManager = Depends(get_db)):
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)